sign_language_datasets/datasets/bsl_corpus/bsl_corpus.py

```diff
@@ -0,0 +1,182 @@
+  year      = {2012},
+  language  = {english},
+  url       = {https://www.sign-lang.uni-hamburg.de/lrec/pub/12033.pdf}
+}
+"""
+
+_HOMEPAGE = "https://bslcorpusproject.org/"
+
+_VIDEO_RESOLUTION = (640, 360)
+
+_FRAMERATE = 25
+
+
+class BslCorpus(tfds.core.GeneratorBasedBuilder):
+    """
+  DatasetBuilder for BSL corpus dataset.
+
+  The dataset currently loads annotations only, and ignores video files.
+  """
+
+    VERSION = tfds.core.Version('1.0.0')
+    RELEASE_NOTES = {
+        # Third release of annotations:
+        # https://bslcorpusproject.org/wp-content/uploads/Notes-to-the-3rd-release-of-BSL-Corpus-annotations.pdf
+        '3.0.0': 'Third release.',
+    }
+
+    BUILDER_CONFIGS = [
+        SignDatasetConfig(name="default", include_video=False, include_pose=None),
+        SignDatasetConfig(name="annotations", include_video=False, include_pose=None),
+    ]
+
+    def __init__(self, bslcp_username: str, bslcp_password: str, **kwargs):
+
+        super(BslCorpus, self).__init__(**kwargs)
+
+        self.bslcp_username = bslcp_username
+        self.bslcp_password = bslcp_password
+
+    def _make_download_manager(self, download_dir, download_config):
+        """Creates a new download manager object."""
+        download_dir = (
+                download_dir or os.path.join(self._data_dir_root, "downloads"))
+        extract_dir = (
+                download_config.extract_dir or os.path.join(download_dir, "extracted"))
+        manual_dir = (
+                download_config.manual_dir or os.path.join(download_dir, "manual"))
+
+        if download_config.register_checksums:
+            # Note: Error will be raised here if user try to record checksums
+            # from a `zipapp`
+            # noinspection PyTypeChecker
+            register_checksums_path = utils.to_write_path(self._checksums_path)
+        else:
+            register_checksums_path = None
+
+        return bsl_corpus_utils.DownloadManagerBslCorpus(
+            download_dir=download_dir,
+            extract_dir=extract_dir,
+            manual_dir=manual_dir,
+            url_infos=self.url_infos,
+            manual_dir_instructions=self.MANUAL_DOWNLOAD_INSTRUCTIONS,
+            force_download=(download_config.download_mode == download.GenerateMode.FORCE_REDOWNLOAD),
+            force_extraction=(download_config.download_mode == download.GenerateMode.FORCE_REDOWNLOAD),
+            force_checksums_validation=download_config.force_checksums_validation,
+            register_checksums=download_config.register_checksums,
+            register_checksums_path=register_checksums_path,
+            verify_ssl=download_config.verify_ssl,
+            dataset_name=self.name,
+            username=self.bslcp_username,
+            password=self.bslcp_password,
+            max_retries=5
+        )
+
+    def _info(self) -> tfds.core.DatasetInfo:
+        """Returns the dataset metadata."""
+        features = {
+            "id": tfds.features.Text(),
```

Comment:
there's also lots of metadata. your call, but i think it might be either worth adding it or at least a comment on its existence